example/gardener-local/kind/cluster-ha-skaffold.yaml

```diff
@@ -0,0 +1,176 @@
+        authorization-webhook-cache-unauthorized-ttl: "0"
+      extraVolumes:
+      - name: gardener
+        hostPath: /etc/gardener/controlplane/auth-webhook-kubeconfig-skaffold.yaml
+        mountPath: /etc/gardener/controlplane/auth-webhook-kubeconfig-skaffold.yaml
+        readOnly: true
+        pathType: File
+  - |
+    apiVersion: kubelet.config.k8s.io/v1beta1
+    kind: KubeletConfiguration
+    maxPods: 500
+    serializeImagePulls: false
+    registryPullQPS: 10
+    registryBurst: 20
+
+- role: worker
+  labels:
+    topology.kubernetes.io/zone: "2"
+  image: kindest/node:v1.24.0
+  extraMounts:
+  - hostPath: example/gardener-local/controlplane
+    containerPath: /etc/gardener/controlplane
+    readOnly: true
+  - hostPath: dev/local-backupbuckets
+    containerPath: /etc/gardener/local-backupbuckets
+  - hostPath: dev/local-registry
+    containerPath: /etc/gardener/local-registry
+  kubeadmConfigPatches:
+  - |
+    kind: ClusterConfiguration
+    apiServer:
+      extraArgs:
+        authorization-mode: RBAC,Node,Webhook
+        authorization-webhook-config-file: /etc/gardener/controlplane/auth-webhook-kubeconfig-skaffold.yaml
+        authorization-webhook-cache-authorized-ttl: "0"
+        authorization-webhook-cache-unauthorized-ttl: "0"
+      extraVolumes:
+      - name: gardener
+        hostPath: /etc/gardener/controlplane/auth-webhook-kubeconfig-skaffold.yaml
+        mountPath: /etc/gardener/controlplane/auth-webhook-kubeconfig-skaffold.yaml
+        readOnly: true
+        pathType: File
+  - |
+    apiVersion: kubelet.config.k8s.io/v1beta1
+    kind: KubeletConfiguration
+    maxPods: 500
+    serializeImagePulls: false
+    registryPullQPS: 10
+    registryBurst: 20
+
+- role: worker
+  labels:
+    topology.kubernetes.io/zone: "3"
+  image: kindest/node:v1.24.0
+  extraMounts:
+  - hostPath: example/gardener-local/controlplane
+    containerPath: /etc/gardener/controlplane
+    readOnly: true
+  - hostPath: dev/local-backupbuckets
+    containerPath: /etc/gardener/local-backupbuckets
+  - hostPath: dev/local-registry
+    containerPath: /etc/gardener/local-registry
+  kubeadmConfigPatches:
+  - |
+    kind: ClusterConfiguration
+    apiServer:
+      extraArgs:
+        authorization-mode: RBAC,Node,Webhook
+        authorization-webhook-config-file: /etc/gardener/controlplane/auth-webhook-kubeconfig-skaffold.yaml
+        authorization-webhook-cache-authorized-ttl: "0"
+        authorization-webhook-cache-unauthorized-ttl: "0"
+      extraVolumes:
+      - name: gardener
+        hostPath: /etc/gardener/controlplane/auth-webhook-kubeconfig-skaffold.yaml
+        mountPath: /etc/gardener/controlplane/auth-webhook-kubeconfig-skaffold.yaml
+        readOnly: true
+        pathType: File
+  - |
+    apiVersion: kubelet.config.k8s.io/v1beta1
+    kind: KubeletConfiguration
+    maxPods: 500
+    serializeImagePulls: false
+    registryPullQPS: 10
+    registryBurst: 20
```

Comment:
Do we need this third pool? We already have the control plane pool with zone `0`, can't we just use it and save a few resources?

Reply:
Previously when we creating a single node kind cluster with worker `role: control-plane`, there is no taint `key:node-role.kubernetes.io/control-plane` because of single node cluster.
```
 kubectl get nodes -o custom-columns=NAME:.metadata.name,TAINTS:.spec.taints --sort-by=.metadata.creationTimestamp
NAME                           TAINTS
gardener-local-control-plane   [map[effect:NoSchedule key:node-role.kubernetes.io/master] map[effect:NoSchedule key:node-role.kubernetes.io/control-plane]]
gardener-local-worker          <none>
gardener-local-worker2         <none>
gardener-local-worker3         <none>
``` 
But when we creating a kind cluster with 4 workers ( one worker with `role: control-plane` + 3 workers with `role:  worker`) ,  `gardener-local-control-plane` node contains taint `key:node-role.kubernetes.io/control-plane` to host only control-plane.
So we need 4 nodes kind cluster to host HA shoot control-plane components.